### PR TITLE
make the rootless code testable

### PR DESCRIPTION
**Pull request by utam0k**
💪
Related: #279

### CODE REVIEW

File: crates/libcontainer/src/rootless.rs [hunk 1]
```diff
@@ -96,6 +97,26 @@ impl<'a> From<&'a Linux> for Rootless<'a> {
     }
 }
 
+#[cfg(not(test))]
```
Comment (yihuaf) [inline crates/libcontainer/src/rootless.rs:100]: This is a super cool idea. Can you add some comments for future us on how this is used in the unit tests?

**Review by yihuaf — APPROVED** [review]
LGTM. A few minor nits and request for more comments.

File: crates/libcontainer/src/rootless.rs (hunk 1 above)
Comment (utam0k) [reply crates/libcontainer/src/rootless.rs:100]: I'm sorry but I couldn't imagine what kind of comments you want 🙇 Please let me more concreate.

File: crates/libcontainer/src/rootless.rs [hunk 2]
```diff
@@ -277,3 +297,204 @@ fn write_id_mapping(
...
+
+    use super::*;
+
+    #[test]
+    fn test_validate_ok() -> Result<()> {
+        let userns = LinuxNamespaceBuilder::default()
+            .typ(LinuxNamespaceType::User)
+            .build()?;
+        let uid_mappings = vec![LinuxIdMappingBuilder::default()
+            .host_id(3333_u32)
```
[the remark this answers is not in the document]
Comment (utam0k) [reply crates/libcontainer/src/rootless.rs]: 👍 fixed

**Review by Furisto — APPROVED** [review]

**Pull request merged**